src/compile.rs

```diff
@@ -514,6 +513,189 @@ fn validate_module_contents<'a>(
     ret
 }
 
+pub fn validate_systems<'a, 'b>(
+    policies: &'b [PolicyFile],
+    module_map: &'b ModuleMap,
+    system_map: &'a mut SystemMap<'b>,
+) -> Result<(), CascadeErrors> {
+    let mut errors = CascadeErrors::new();
+
+    // Store all systems across files in a vector
+    let mut systems_vec: Vec<(&SimpleFile<String, String>, &System)> = Vec::new();
+    for p in policies {
+        for e in &p.policy.exprs {
+            if let Expression::Decl(Declaration::System(s)) = e {
+                systems_vec.push((&p.file, s));
+            }
+        }
+    }
+
+    for (file, system) in &systems_vec {
+        let mut system_modules = BTreeSet::new();
+        let mut configs = BTreeMap::new();
+
+        // Validate that the modules of a system exist
+        for m in &system.modules {
```

Comment:
I feel like declaring a system with no modules should be an error.  Are we checking that anywhere?